src/cert_store.rs

```diff
@@ -520,66 +522,67 @@ mod test {
                     break;
                 }
             }
+            let mut identity = identity.unwrap();
 
-            let mut identity = match identity {
-                Some(identity) => identity,
-                None => {
-                    panic!()
-                }
-            };
-
+            // open temporary "TestRustMy" store for current user and add identity if it is not already in the store
             let mut store = CertStore::open_current_user("TestRustMy").unwrap();
-            if let Err(_err) = CertStore::find_existing_cert(&mut store, &identity) {
-                if let Err(_err) = store.add_cert(&identity, CertAdd::Always){
-                    panic!()
-                }
+            if CertStore::find_existing_cert(&mut store, &identity).is_err() {
+                store.add_cert(&identity, CertAdd::Always).unwrap();
                 continue;
             }
 
+            // the following code will only be executed on second iteration
             let existing_cert = CertStore::find_existing_cert_and_key(&mut identity, &mut store).unwrap();
 
-            let mut exisiting_cert = match existing_cert {
-                Some(exisiting_cert) => exisiting_cert,
-                None => {
-                    panic!()
-                }
-            };
-
-            assert_eq!(identity.private_key().silent(true).compare_key(true).acquire().is_err(), true);
+            // get handle to current cert inside of "TestRustMy" store
+            let mut exisiting_cert = existing_cert.unwrap();
```

Comment:
In the line above: let existing_cert = CertStore::find_existing_cert_and_key(&mut identity, &mut store).unwrap();

You are already doing the existing_cert.unwrap(); Not sure why you need to do again?

Reply:
call signature for find_existing_cert_and_key:

pub fn find_existing_cert_and_key(provided_cert: &mut CertContext, store: &mut CertStore) -> io::Result<Option<CertContext>>

return type is <Option<CertContext>>

First unwrap is to get inner value of result type, Some() or None().

Second unwrap is to get the inner value of the Option.

The second unwrap is not needed, will change to:

let mut existing_cert = CertStore::find_existing_cert_and_key(&mut identity, &mut store).unwrap().unwrap();